tests/core_tests/test_context.py

```diff
@@ -1140,6 +1155,25 @@ def test_serialize_without_user(self):
         tank.Context.deserialize(ctx_str)
         self._assert_same_user(tank.get_authenticated_user(), other_user)
 
+    def test_serialize_to_dict(self):
+        """
+        Make sure a context serialized to a dictionary can be unserialized.
+        """
+        ctx = context.Context(**self.kws)
+        ctx_dict = ctx.to_dict()
+        new_ctx = tank.Context.from_dict(ctx.sgtk, ctx_dict)
+        self._assert_equal_contexts(new_ctx, ctx)
+
+    def _assert_equal_contexts(self, ctx_1, ctx_2):
+        """
+        Ensures two contexts are equal.
+        Note that source_entity is not part of the equality comparison,
```

Comment:
@thebeeland Do you remember why that is not the case? It should probably be documented in the __eq__ method.